src/sentry/snuba/discover.py

```diff
@@ -894,24 +899,40 @@ def histogram_query(
     )
 
     key_column = None
+    array_column = None
     conditions = []
     if len(fields) > 1:
-        key_column = "array_join(measurements_key)"
-        key_alias = get_function_alias(key_column)
-        measurements = []
-        for f in fields:
-            measurement = get_measurement_name(f)
-            if measurement is None:
-                raise InvalidSearchQuery(f"multihistogram expected all measurements, received: {f}")
-            measurements.append(measurement)
-        conditions.append([key_alias, "IN", measurements])
+        if all(is_measurement(field) for field in fields):
+            key_column = "array_join(measurements_key)"
+            key_alias = get_function_alias(key_column)
+            array_column = "measurements"
+            measurements = []
+            for f in fields:
+                measurement = get_measurement_name(f)
+                measurements.append(measurement)
+            conditions.append([key_alias, "IN", measurements])
+        elif all(is_span_op_breakdown(field) for field in fields):
+            key_column = "array_join(span_op_breakdowns_key)"
+            key_alias = get_function_alias(key_column)
+            array_column = "span_op_breakdowns"
+            breakdowns = []
+            for f in fields:
+                span_op_breakdown = get_span_op_breakdown_name(f)
+                breakdowns.append(span_op_breakdown)
+            conditions.append([key_alias, "IN", breakdowns])
```

Comment:
🏌️ something roughly like this:
```suggestion
       histogram_type = check_histogram_fields(fields)
        if histogram_type == "measurements"
            key_column = "array_join(measurements_key)"
            array_column = "measurements"
            histogram_function = get_measurement_name
        elif histogram_type == "op_breakdown"
            key_column = "array_join(span_op_breakdowns_key)"
            array_column = "span_op_breakdowns"
            histogram_function = get_span_op_breakdown_name
        else:
            raise ...
       
       if histogram_type:
           key_alias = get_function_alias(key_column)
           field_names = [histogram_function(field) for field in fields]
           conditions.append([key_alias, "IN", field_names])
```